scripts/image_compression.py

```diff
@@ -0,0 +1,127 @@
+
+
+def check_compressable_images(
+        input_path: Union[str, pathlib.Path]
+    ) -> List[CompressedImageInfo]:
+    """Check and compress images using GraphicsMagick."""
+
+    result_images: List[CompressedImageInfo] = []
+    supported_extensions = {'.png', '.jpg', '.jpeg', '.webp'}
+
+    for file_path in pathlib.Path(input_path).glob('**/*.*'):
+        if file_path.suffix.lower() not in supported_extensions:
+            continue
+
+        try:
+            with Image.open(file_path):
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    temp_compressed = (
+                        pathlib.Path(tmpdir) / f'compressed_{file_path.name}'
+                    )
+
+                    if file_path.suffix.lower() == '.png':
+                        cmd = [
+                            'gm', 'convert',
+                            str(file_path),
+                            '-strip',
+                            '-compress', 'Zip',
+                            str(temp_compressed)
+                        ]
+                    elif file_path.suffix.lower() in {'.jpg', '.webp'}:
+                        cmd = [
+                            'gm', 'convert',
+                            str(file_path),
+                            '-strip',
+                            '-compress', 'LZW',
+                            str(temp_compressed)
+                        ]
+
+                    result = subprocess.run(
+                        cmd, capture_output=True, text=True, check=False
+                        )
+                    if result.returncode == 0 and temp_compressed.exists():
+                        original_size = file_path.stat().st_size
+                        new_size = temp_compressed.stat().st_size
+
+                        if new_size < original_size * 0.99:
+                            result_images.append({
+                                'path': file_path,
+                                'original_size': original_size,
+                                'new_size': new_size
+                            })
+                    else:
+                        logging.info(
+                            'Compressed image > original image'
+                        )
+                        continue
+        except Exception as e:
+            print(f'[ERROR] Could not process {file_path}: {e}')
+    return result_images
+
+
+def main() -> None: # pragma: no cover
+    """Main function to compress images in the repository."""
+    repo_path = pathlib.Path('./assets')
+    compressed_images = check_compressable_images(str(repo_path))
+
+    if compressed_images:
+        space = 0
+        print(len(compressed_images), 'images could be compressed further:\n')
+        for image in compressed_images:
+            print(image['path'])
+            saved = image['original_size'] - image['new_size']
+            space += saved
+        print('Use the following command to compress the images:\n')
```

Comment:
You can print this as a backup, but don't rely on it; contributors don't have gm installed on their system and that's going to potentially introduce a whole bunch of installation problems.

Instead, create the files and upload them as a workflow artifact (where the image filenames are based on their codebase paths, perhaps) so that they can be downloaded. Give instructions here on how the contributor would download the images and replace them in their PR.